### PR TITLE
AWS - Step Functions - Activities Resource & Tagging support

### LINKED ISSUE #9695: aws - step function - activities support
### Describe the feature

It would be great for c7n to support Step Function Activities.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-stepfunctions-activity.html
https://awscli.amazonaws.com/v2/documentation/api/latest/reference/stepfunctions/describe-activity.html

### Extra information or context

_No response_

### PR DESCRIPTION
**Pull request by zkarpinski**
Adds support for Step Function Activity resource tagging & KMS filtering

Resolves #9695 & #9696

### CODE REVIEW

**Comment by kapilt** [comment]
Could you add some example policies re doc strings, it’s not clear why this is useful else. Neither list/get on activities seem to have useful info.

**Comment by zkarpinski** [comment]
Thanks @kapilt updated with 2 examples in string docs also added KMS filtering functionality.

File: c7n/resources/sfn.py [hunk 1]
```diff
@@ -40,6 +41,59 @@ class resource_type(TypeInfo):
...
 
 
+class DescribeActivity(DescribeSource):
+
+    def augment(self, resources):
+        resources = super().augment(resources)
+        return universal_augment(self.manager, resources)
+
+
+@resources.register('activity')
```
Comment (PratMis) [inline c7n/resources/sfn.py:51]: The name of the resource here lacks context. Would it make sense to rename it to something that contains sfn?

Comment (zkarpinski) [reply c7n/resources/sfn.py:51]: renaming to sfn-activity

**Review by PratMis — APPROVED** [review]
lgtm, thanks

**Pull request merged**